sssd_test_framework/utils/pam.py

```diff
@@ -0,0 +1,278 @@
+        :type host: MultihostHost
+        :param fs: Linux file system instance
+        :type fs: LinuxFileSystem
+        """
+        super().__init__(host)
+
+        self.fs: LinuxFileSystem = fs
+
+    def access(self, file: str = "/etc/security/access.conf") -> PAMAccess:
+        """
+        :param file: PAM Access file name.
+        :type file: str
+        :return: PAM Access object
+        :rtype: PAMAccess
+        """
+        return PAMAccess(self, file, self.host, self.fs)
+
+    def faillock(self, file: str = "/etc/security/faillock.conf") -> PAMFaillock:
+        """
+        :param file: PAM Faillock file name.
+        :type file: str
+        :return: PAM Faillock object
+        :rtype: PAMFaillock
+        """
+        return PAMFaillock(self, file, self.host, self.fs)
+
+
+class PAMAccess(PAMUtils):
+    """
+    Management of PAM Access on the client host.
+
+    .. code-block:: python
+        :caption: Example usage
+
+        @pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+        def test_example(client: Client, provider: GenericProvider):
+            # Add users
+            provider.user("user-1").add()
+            provider.user("user-2").add()
+
+            # Add rule to permit "user-1" and deny "user-2"
+            access = client.pam.access()
+            access.config_set([{"access": "+", "user": "user-1", "origin": "ALL"},
+                               {"access": "-", "user": "user-2", "origin": "ALL"}])
+            access.config_write()
+
+            client.sssd.authselect.enable_feature(["with-pamaccess"])
+            client.sssd.start()
+
+            # Check the results
+            assert client.auth.ssh.password("user-1", "Secret123")
+            assert not client.auth.ssh.password("user-2", "Secret123")
+    """
+
+    def __init__(self, util: PAMUtils, file: str, host: MultihostHost, fs: LinuxFileSystem) -> None:
+        """
+        :param util: PAMUtils utility object
+        :type util: PAMUtils
+        :param file: Configuration file
+        :type file: str
+        :param host: Multihost object
+        :type host: MultihostHost
+        :param fs: LinuxFileSystem object
+        :type fs: LinuxFileSystem
+        :param file: File name of access file
+        :type file: str, optional
+        """
+        super().__init__(host, fs)
+        self._changed: bool = False
+
+        self.util: PAMUtils = util
+        self.file: str = file
+        self.path: str = "/files" + self.file
+        self.args: str = f'--noautoload --transform "Access.lns incl {self.file}"'
+        self.cmd: str = ""
+
+        self._backup = f"{self.file}.bak.wfw"
+
+    def setup_when_used(self):
+        super().setup_when_used()
+        self.fs.backup(self.file)
+
+    def config_write(self):
+        """
+        Write access configuration.
+        :return: None
+        """
+        self.util.logger.info(f"Saving Augeas changes to {self.file} on {self.util.host.hostname}")
```

Comment:
"on host" part can now be omitted, since https://github.com/next-actions/pytest-mh/commit/d5431c160e2d83e3dbfb400ff13eef36d0685479 (applies everywhere)

Reply:
ack, updated the message.